### ISSUE TITLE
First bare-metal boot fails to mount /run/media/iso on 32.20200601.3.0, 31.20200517.3.0 and 31.20200505.3.0

### ISSUE THREAD

**Issue by AlexGustafsson** [coreos/fedora-coreos-tracker#554]
I'm trying to install from a live ISO on bare metal.

When first booting from a freshly created USB of the 32.20200601.3.0, 31.20200517.3.0 or 31.20200505.3.0 live ISO, the boot fails to mount /run/media/iso. 31.20200420.3.0 is the latest ISO that I've found works.

```
systemd[1]: Mounting /run/media/iso
systemd-journald[687]: Missed 2 kernel messages
kernel: ISOFS: Unable to identify CD-ROM format.
mount[692]: mount /run/media/iso: wrong fs type, bad option, bad superblock on /dev/sdb1, missing codepage or helper program or other error.
systemd[1]: run-media-iso.mount: Failed with result 'exit-code'.
systemd[1]: Failed to mount /run/media/iso
```

The USB was flashed using Rufus on Windows, with which I've never had an issue before. I've created USBs for Fedora CoreOS using Rufus in the past.

Settings used for flashing:

* Partitioning scheme: MBR
* File system: FAT32
* Cluster size: 16KB
* Write in "ISO mode" as opposed to "DD mode"

Worth mentioning is that when I manually run `mount /dev/sdb1 /run/media/iso`, it works and I'm able to view its contents.

Is this something anyone is able to replicate?
What has changed in the ISO handling since 31.20200420.3.0 that could affect this?

**Comment by lucab** [comment]
Thanks for the report.

> Write in "ISO mode" as opposed to "DD mode"

I have a feeling that from the point of view of the burning utility this may mean "let me apply some smart optimizations to the image". Does the other mode work better? I would expect the latter to be closer to "dumbly write the content without manipulating it".

**Comment by dustymabe** [comment]
> What has changed in the ISO handling since 31.20200420.3.0 that could affect this?

One thing that changed is that we started [splitting the root filesystem out of the initramfs](https://github.com/coreos/coreos-assembler/pull/1420) so it wouldn't take so long to load everything into memory.

**Comment by MarkIannucci** [comment]
I ran into a similar problem when I wrote the USB key in `ISO Mode`.  When I retried using `DD mode`, it worked.

**Comment by dustymabe** [comment]
@MarkIannucci since you've found a way forward I'm going to close out this ticket. I assume the tool is doing some enhancements when using `ISO Mode` which doesn't play nice with the way we've set up the image. `DD mode` for the win!

**Comment by AlexGustafsson** [comment]
I haven’t had the time to try out the DD mode. But I think you’re completely right. There’s nothing wrong with CoreOS itself.

**Comment by dustymabe** [comment]
oops @AlexGustafsson - I mistakenly thought @MarkIannucci opened the ticket and was reporting he found success.

If you'd like me to re-open I'll be glad to do so while you investigate further.

**Comment by AlexGustafsson** [comment]
No worries! I’m sure @MarkIannucci is right. I’ll see if I get the time to try the DD mode. But for now I think it’s safe to keep this closed.

**Comment by bnordgren** [comment]
I think this should be re-opened. Yes, I'm sure "dd" mode works because the failure is that the mount unit in systemd is looking for an ISO9660 image. When you DD it, you get an ISO image. Ironically, using that Rufus tool, "ISO" mode actually means "format the USB stick normally and copy all the files over", so the result is a vfat formatted stick, not an ISO9660. 

Prior to this weekend I knew nothing about Rufus. I found out about it because my Dell PowerEdge T310 would not boot a USB stick created with the "dd" command line. Or with the Fedora Media Writer, which does the same thing, as far as I can tell. However, a vfat formatted USB which is bootable will be recognized. (Note: The same boot media--using the dd method--works fine in a normal PC, just not on the one I'm trying to install.) Note that this Rufus tool (in ISO mode) is on the Dell website as the tool to use to make bootable USBs for OS installation on their PowerEdge servers.

So, while "dd" works with the CoreOS image, it is not always possible to boot using that method; ergo the CoreOS image needs to support mounting /run/media/iso as "whatever it is", not locking it down to ISO9660. Or write off certain server class machines from certain manufacturers as "not supported". 

I'm super annoyed with Dell for making me use Windows to install Linux. Fedora media writer should probably offer similar functionality as this "Rufus" tool, as well, but that's a different ticket for a different project.

**Comment by bdlink** [comment]
On Nov 29, 2020, at 7:42 AM, bnordgren <notifications@github.com> wrote:

﻿
CAUTION: This email originated from outside of BCIT. Do not click links or open attachments unless you recognize the sender and know the content is safe.


I think this should be re-opened. Yes, I'm sure "dd" mode works because the failure is that the mount unit in systemd is looking for an ISO9660 image. When you DD it, you get an ISO image. Ironically, using that Rufus tool, "ISO" mode actually means "format the USB stick normally and copy all the files over", so the result is a vfat formatted stick, not an ISO9660.

Prior to this weekend I knew nothing about Rufus. I found out about it because my Dell PowerEdge T310 would not boot a USB stick created with the "dd" command line. Or with the Fedora Media Writer, which does the same thing, as far as I can tell. However, a vfat formatted USB which is bootable will be recognized. (Note: The same boot media--using the dd method--works fine in a normal PC, just not on the one I'm trying to install.) Note that this Rufus tool (in ISO mode) is on the Dell website as the tool to use to make bootable USBs for OS installation on their PowerEdge servers.

So, while "dd" works with the CoreOS image, it is not always possible to boot using that method; ergo the CoreOS image needs to support mounting /run/media/iso as "whatever it is", not locking it down to ISO9660. Or write off certain server class machines from certain manufacturers as "not supported".

I'm super annoyed with Dell for making me use Windows to install Linux. Fedora media writer should probably offer similar functionality as this "Rufus" tool, as well, but that's a different ticket for a different project.

**Comment by dustymabe** [comment]
> I found out about it because my Dell PowerEdge T310 would not boot a USB stick created with the "dd" command line.

That seems like a problem. Can you open a new issue with more details?

**Comment by TiloGit** [comment]
fyi had same today with RH OpenShift 4.6 (Install OpenShift on Bare Metal with the Assisted Installer).
 Change the rufus write mode to DD fix it. Thx

**Comment by michelcve** [comment]
I have the same problem when booting from USB. The mount command includes something like '-t cdrom' or something, which doesn't really work with USB sticks.

If you have console access, this is what I do to work around the problem (basically mounting it myself):

`mount /dev/sda1 /run/media/iso/`
`rm /run/systemd/generated/run-media-iso.mount`
`init 3`

Replace `/dev/sda1` with the partition of your USB stick if it is named differently then mine.

Basically you do the mounting yourself, remove the script that attempts to do the mounting, and restart the boot process (in part). I should get you to the prompt you're expecting, and you can run `coreos-installed` without a problem from that point on.


Though this workaround does help, it would still be far better if the mount command could be corrected to work for USB devices as well :-)

**Comment by bgilbert** [comment]
@michelcve Booting from USB is tested by our CI, so this is supposed to work properly.  Please open a new bug with the details of the error you're seeing.

This is an old bug that continues to collect unrelated issues, so I'm locking it.  Please open a new bug if you're having boot problems with the ISO.
